SPI component compile error in rp2040 / rpipico

### The problem

I can't seem to compile a RP2040 firmware with an SPI component on 2026.2.4 due to compiler errors related to SPI

### Which version of ESPHome has the issue?

2026.2.4

### What type of installation are you using?

pip

### What platform are you using?

RP2040

### Component causing the issue

spi

### YAML Config

```yaml
esphome:
  name: test

rp2040:
  board: rpipico

spi:
  - id: spi0
    clk_pin: GPIO2
    mosi_pin: GPIO3
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2026.2.4
INFO Reading configuration test.yaml...
INFO Generating C++ source...
INFO Compiling app... Build path: /home/nitroxis/esphome/.esphome/build/test
Processing test (board: rpipico; framework: arduino; platform: https://github.com/maxgerhardt/platform-raspberrypi.git#v1.2.0-gcc12)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HARDWARE: RP2040 133MHz, 256KB RAM, 2MB Flash
 - framework-arduinopico @ 1.30904.0 (3.9.4) 
 - tool-picotool-rp2040-earlephilhower @ 5.120300.240818 (12.3.0) 
Flash size: 2.00MB
Sketch size: 1.00MB
Filesystem size: 1.00MB
Maximium Sketch size: 1044480 EEPROM start: 0x101ff000 Filesystem start: 0x100ff000 Filesystem end: 0x101ff000
Dependency Graph
|-- SPI @ 1.0
Compiling .pioenvs/test/src/esphome/components/spi/spi.cpp.o
Compiling .pioenvs/test/src/esphome/components/spi/spi_arduino.cpp.o
Compiling .pioenvs/test/src/esphome/core/application.cpp.o
Compiling .pioenvs/test/src/esphome/core/component.cpp.o
Compiling .pioenvs/test/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/test/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/test/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/test/src/esphome/core/util.cpp.o
Compiling .pioenvs/test/src/main.cpp.o
In file included from /home/nitroxis/.platformio/packages/framework-arduinopico/libraries/SPI/src/SPI.h:25,
                 from src/esphome/components/spi/spi.h:18,
                 from src/esphome.h:7,
                 from src/main.cpp:3:
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:55:27: error: expected ')' before '*' token
   55 | #define spi0 ((spi_inst_t *)spi0_hw)
      |               ~           ^
src/main.cpp:10:27: note: in expansion of macro 'spi0'
   10 | static spi::SPIComponent *spi0;
      |                           ^~~~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:55:27: error: expected ')' before '*' token
   55 | #define spi0 ((spi_inst_t *)spi0_hw)
      |              ~            ^
src/main.cpp:10:27: note: in expansion of macro 'spi0'
   10 | static spi::SPIComponent *spi0;
      |                           ^~~~
src/main.cpp: In function 'void setup()':
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:55:15: error: lvalue required as left operand of assignment
   55 | #define spi0 ((spi_inst_t *)spi0_hw)
      |              ~^~~~~~~~~~~~~~~~~~~~~~
src/main.cpp:84:3: note: in expansion of macro 'spi0'
   84 |   spi0 = new spi::SPIComponent();
      |   ^~~~
src/main.cpp:85:7: error: invalid use of incomplete type 'spi_inst_t' {aka 'struct spi_inst'}
   85 |   spi0->set_component_source(LOG_STR("spi"));
      |       ^~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:47:16: note: forward declaration of 'spi_inst_t' {aka 'struct spi_inst'}
   47 | typedef struct spi_inst spi_inst_t;
      |                ^~~~~~~~
src/main.cpp:90:7: error: invalid use of incomplete type 'spi_inst_t' {aka 'struct spi_inst'}
   90 |   spi0->set_clk(rp2040_rp2040gpiopin_id);
      |       ^~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:47:16: note: forward declaration of 'spi_inst_t' {aka 'struct spi_inst'}
   47 | typedef struct spi_inst spi_inst_t;
      |                ^~~~~~~~
src/main.cpp:94:7: error: invalid use of incomplete type 'spi_inst_t' {aka 'struct spi_inst'}
   94 |   spi0->set_mosi(rp2040_rp2040gpiopin_id_2);
      |       ^~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:47:16: note: forward declaration of 'spi_inst_t' {aka 'struct spi_inst'}
   47 | typedef struct spi_inst spi_inst_t;
      |                ^~~~~~~~
src/main.cpp:95:7: error: invalid use of incomplete type 'spi_inst_t' {aka 'struct spi_inst'}
   95 |   spi0->set_interface(&SPI);
      |       ^~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:47:16: note: forward declaration of 'spi_inst_t' {aka 'struct spi_inst'}
   47 | typedef struct spi_inst spi_inst_t;
      |                ^~~~~~~~
src/main.cpp:96:7: error: invalid use of incomplete type 'spi_inst_t' {aka 'struct spi_inst'}
   96 |   spi0->set_interface_name("SPI");
      |       ^~
/home/nitroxis/.platformio/packages/framework-arduinopico/pico-sdk/src/rp2_common/hardware_spi/include/hardware/spi.h:47:16: note: forward declaration of 'spi_inst_t' {aka 'struct spi_inst'}
```

### Additional information

_No response_

--- a/esphome/config_validation.py
+++ b/esphome/config_validation.py
@@ -244,6 +244,8 @@ RESERVED_IDS = [
     "open",
     "setup",
     "loop",
+    "spi0",
+    "spi1",
     "uart0",
     "uart1",
     "uart2",

